### ISSUE TITLE
Hud indicators current channel mode 

### ISSUE THREAD

**Issue by Klaranth** [pandorabox-io/in-game#288]
SX
postool beerchat extension to get hud indicators for current channel / active whisper mode /cc SwissalpS
Also selected @ message target :) ^^

**Comment by S-S-X** [comment]
This might be better done in beerchat, postool for information hud configuration just happens to already contain somewhat similar things and on pandorabox it happens to be well known user interface for basic hud configuration.

**Comment by SwissalpS** [comment]
I agree with S-S-X, this is something for beerchat. Maybe /beerchat_hud opens a formspec similar to /postool so players can configure. If needed at all, probably a toggle is sufficient.
The dificult part is figuring out where to place yet another hud.

**Comment by S-S-X** [comment]
> If needed at all, probably a toggle is sufficient.

Toggle is probably bad idea because there's large number of requests for beerchat in game configuration, actually I'm pretty sure that single formspec wont be enough (also reason why I've not already added configuration chat commands for beerchat):
- https://github.com/minetest-beerchat/beerchat/issues/12
- https://github.com/minetest-beerchat/beerchat/issues/14
- https://github.com/minetest-beerchat/beerchat/issues/49
- multiple other more or less related management/configuration things that would be better implemented with formspecs: channel bans, channel aliases, extension configurations, channel members, remote bans, all other sounds, whatnot.

Beerchat would really benefit from generic (tabbed) formspec config API that allows adding configuration pages and few predefined configuration elements: text, number, boolean.

I've been planning that for some time (...oh crap yes ...seems that one issue is >2y old already) and have done few simple tests but do not have anything really usable yet. Well metatool has but it is unfinished, bad API spec, bad coordinate handling and not simple/restricted enough.

> The dificult part is figuring out where to place yet another hud.

For HUD position I think it might be possible to check, use and possibly even adjust postool HUD to seamlessly extend it with beerchat HUD if postool is installed. That would work at least for pandorabox.

**Comment by OgelGames** [comment]
Could just add a command that tells you all the information at once, something like `/chatstatus`.

Really there only needs to be *some* way of checking, without actually sending a chat message to test it. I don't see why you would want it constantly on your screen...

**Comment by SwissalpS** [comment]
/mc already outputs current default channel at bottom of list, this 'flushes' your chat though if you are joined to a lot of channels.

**Comment by S-S-X** [comment]
> Could just add a command that tells you all the information at once, something like /chatstatus.

This won't really be that useful for this issue, you could add that but it sure wont reduce requests for basic feature that does not require typing commands. There's already simpler/shorter workarounds and many people know about it.

> Really there only needs to be _some_ way of checking, without actually sending a chat message to test it. I don't see why you would want it constantly on your screen...

Just some way of checking is not good enough. There is way to ensure it by just writing `#channel`.
Reason why this feature is requested or mentioned by many people is simply that current workarounds always require you to type something.

Using any other chat platform you will immediately know what you're sending and where it will go, in Minetest with chat mods adding channels this not possible because engine does not provide enough features to do this nicely.

Next best option for people who do not want to type some command just to see their chat status is to have status in HUD, that allows you to see your own status at any time without using chat commands.

Everyone knows that this proposal is not best UX but I don't think there's any better options available with current engine features.

**Comment by S-S-X** [comment]
Going to add this as a beerchat extension which allow trying it out and deciding if hud should stay available or not.
Another good thing anyway is that working on this also helps with multiple other requested beerchat features.
